interview-questions.md

```diff
@@ -8,64 +8,86 @@ Then, research the question to expand on your answer. Even if you feel you have
 
   Your answer:
+  CRUD stands for Create, Read, Update, and Delete. CRUD is the four basic things you want APIs that you build to do. Create data, read data, update existing data, and destroy data.
 
-  Researched answer:
-
+  Researched answer:  
+  Expanding on my answer, a CRUD application uses forms to get data into and out of databases. It also provides developers a framework to follow in order to create full usable models.
 
 
 3. What is a migration? Why would you use one?
 
   Your answer:
+  A migration is our way to update our rails schema. In rails you never want to update the schema directly since there are a lot of behind the scenes processes that we don't see. migrations allow us to update the schema safely without breaking the app.
 
   Researched answer:
-
+  expanding on my own answer, it also allows other developers working on the same project to update their schema if someone else made any changes, it also allows you to bring databases up to date whenever you roll out a new release. In general migrations just help keep things more synchronized across different machines/workstations.
 
 
 4. Describe as many possible relationships between a model called Person and a model called Computer as you can think of. Based on each description, which model would hold the foreign key?
 
   Your answer:
+  I think it kind of depends on the situation.
+
+  A person like me has_many computers.
+  My personal computer, my personal laptop, LEARN's laptop.
+
+  But if we look at it from the loaner laptop, a computer has_many persons using it. the person that had it before me, me, and the one that will use it after me.
+
+  depending on the situation the model that will hold the foreign key is the model that belongs to another model.
 
   Researched answer:
+  Looking back onto the syllabus, only other way I could think of is that a person can has one through if there was a specific way the person came to have that computer.
 
 
 
 5. What is object-relational mapping?
 
   Your answer:
+  It's a way for us to look into our database in tables. It also allows us to see the relationships between different models in these tables.
 
   Researched answer:
+  It's a programming technique that converts information from incompatible types into objects that we can use. (using object oriented languages like ruby) This allows us to make queries using the programming language used.
 
 
 
 6. What is a gem?
 
   Your answer:
-
+A gem is a ruby package or library that people can download install and use.
   Researched answer:
-
+It is a standard format for programmers to distribute ruby programs and libraries. 
 
 
 7. Why is it important to have validations in your application?
 
   Your answer:
+  It is important to have validations in your apps to make sure that potential users enter required data that we need, and to make sure that the data matches our specifications/requirements.
```

Comment:
Absolutely. Additionally I would mention how it helps provide some simple security to ensure that the data input cannot harm the database.